flopy/utils/geometry.py

```diff
@@ -898,46 +898,80 @@ def project_point_onto_xc_line(line, pts, d0=0, direction="x"):
     pts : list or np.ndarray
         numpy array of [(x, y),] points to be projected
     d0 : distance offset along line of min(xl)
-    direction : string
-        projection direction "x" or "y"
+    calc_dist : bool
+        boolean flag to indicate that the return type is distance offset
+        by d0 (calculated from line[0]). If false this will return the
+        "projected" xy location along the cross-sectional line
+
 
     Returns:
-        np.ndarray of projected [(x, y),] points
+        tuple of (x , y) or distance
     """
     if isinstance(line, list):
         line = np.array(line)
 
     if isinstance(pts, list):
         pts = np.array(pts)
+        if pts.ndim == 1:
+            pts = np.expand_dims(pts, axis=0)
 
     x0, x1 = line.T[0, :]
     y0, y1 = line.T[1, :]
-    dx = np.abs(x0 - x1)
-    dy = np.abs(y0 - y1)
+    dx = x1 - x0
+    dy = y1 - y0
+    if dx == 0:
```

Comment:
is the exact zero check sufficient or would approximate equality be safer? to avoid not only inf but unrepresentably large slope (greater than `sys.float_info.max`)

Reply:
The exact zero check is only to avoid an "inf" condition in the case of a vertical line, which is a common occurance (e.g., `line={"column": 1}`). In this case we want to replace `dx=0` with `dx=very small number` to be able to approximate the calculation.